kapitan/cli.py

```diff
@@ -276,7 +280,7 @@ def _search_imports(cwd, imp):
         compile_targets(args.inventory_path, search_paths, args.output_path,
                         args.parallelism, args.targets, ref_controller,
                         prune=(args.prune), indent=args.indent, reveal=args.reveal,
-                        cache=args.cache, cache_paths=args.cache_paths,
+                        cache=args.cache, cache_paths=args.cache_paths, dependency_cache_path=args.dependency_cache_path, force_fetch=args.fetch,
```

Comment:
@uberspot @ramaro  caching is still done for git repositories since we have subdir and/or ref attributes that may make the actual directories different among different targets. For example, if we have a git dependency with subdir, we will still want to retain the entire repository for other targets to use other subdirs in the repository. We could of course make this caching temporary and fetch the repository every time the subdir or ref changes. Let me know what you think.

Reply:
sgtm, it's much clearer now as a process.

Reply:
Wouldn't fetching the whole repo and dealing with whatever subdir a target might need externally be simpler? That way the "cache" is the downloaded repo itself.
I propose we do something like:
 * The first time a repo is needed for a target (e.g. files no found), download it entirely
 * If another target needs a different subdir of that repo, the user should override it in the inventory for that target.

Or am I failing to understand the need for a dedicated cache dir here?

Reply:
@ramaro thanks for the comment.
I'm just not sure where to save the entire repo in the way that is common across targets such that it can be reused as cache across targets. Say we have a target that has a git dependency with subdir specified, I believe the `output_path` should contain the subdir extracted from the repo. Then where should the entire cloned repo be saved? Or following your approach,

> The first time a repo is needed for a target (e.g. files no found), download it entirely

So in this case, the entire repo is first saved to the `output_path` as given in this target.

> If another target needs a different subdir of that repo, the user should override it in the inventory for that target.

For this target, the `output_path` is different because it wants the subdir to be at `output_path`, which is fine. But then how should we know where the `output_path` to the entire repo is during compilation of this target in order to reuse it as cache?

I could totally be misunderstanding your approach. Please let me know if that's the case!